plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsViewModel.java

```diff
@@ -92,7 +95,11 @@ protected void update(Connection connection, List<Connection> connections,
 		}
 		deployProject = updateDeployProject(deployProject, projects);
 		updateSourcePath(sourcePath, deployProject);
-		updatePodPath(podPath, deploymentMapperByProjectName, service);
+		if(serviceOrDeploymentChanged) {
```

Comment:
if path is null we probably want to initialize it from deploymentMapperByProjectName & service